File: javascript/hamming/hamming.js
function Hamming () {}

Hamming.prototype.compute = function (strand1, strand2) {
  var difference = 0;

  if (strand1.length !== strand2.length) throw new Error('DNA strands must be of equal length.');

  for (var i = 0; i < strand1.length; i++) {
    if (strand1.charAt(i) !== strand2.charAt(i)) difference++;
  }
  return difference;
};

module.exports = Hamming;

File: ruby/hamming/hamming.rb
class Hamming

  def self.compute (strand1, strand2)
    dna1 = strand1.split('')
    dna2 = strand2.split('')
    index = 0
    difference = 0

    if dna1.length != dna2.length
      raise (ArgumentError)
    end

    while index < dna1.length
      difference += 1 if dna1[index] != dna2[index]
      index += 1
    end

    difference
  end

end
